services/merak-topo/handler/handler.go

```diff
@@ -242,174 +330,326 @@ func UpdateComputenodeInfo(client *kubernetes.Clientset, topo_id string, returnM
-					mac = s2
-					count = count + 1
-					if count == 2 {
-						database.SetValue(topo_id+":"+ip, mac)
-						count = 0
-					}
-				}
+			crm.Id = cnode.Id
+			crm.Name = cnode.Name
+			crm.DatapathIp = cnode.DatapathIp
+			crm.ContainerIp = cnode.ContainerIp
+			crm.Mac = cnode.Mac
+			crm.Veth = cnode.Veth
+			if cnode.Status == database.STATUS_READY {
+				crm.Status = pb_common.Status_READY
+			} else if cnode.Status == database.STATUS_DELETING {
+				crm.Status = pb_common.Status_DELETING
+			} else if cnode.Status == database.STATUS_DEPLOYING {
+				crm.Status = pb_common.Status_DEPLOYING
+			} else {
+				crm.Status = pb_common.Status_NONE
 			}
-			if count != 0 {
-				log.Printf("arp output parsing error. ip and mac doesn't match")
+			if cnode.OperationType == database.OPERATION_CREATE {
+				crm.OperationType = pb_common.OperationType_CREATE
+			} else if cnode.OperationType == database.OPERATION_INFO {
+				crm.OperationType = pb_common.OperationType_INFO
+			} else if cnode.OperationType == database.OPERATION_DELETE {
+				crm.OperationType = pb_common.OperationType_DELETE
+			} else if cnode.OperationType == database.OPERATION_UPDATE {
+				crm.OperationType = pb_common.OperationType_UPDATE
 			}
+
+			returnMessage.ComputeNodes = append(returnMessage.ComputeNodes, &crm)
+
 		}
+
 	}
+
+	go UpdateComputenodeInfo(k8client, topo_id)
+
 	return nil
 }
 
-func QueryHostNode(k8client *kubernetes.Clientset, topo_id string) error {
+// func makehrm(node database.HostNode, hrm pb_common.InternalHostInfo) {
+
+// 	hrm.Ip = node.Ip
+// 	if node.Status == database.STATUS_READY {
+// 		hrm.Status = pb_common.Status_READY
+// 	} else if node.Status == database.STATUS_DELETING {
+// 		hrm.Status = pb_common.Status_DELETING
+// 	} else {
+// 		hrm.Status = pb_common.Status_NONE
+// 	}
+
+// 	hrm.RoutingRules = node.Routing_rule
+
+// }
+
+// func makecrm(node database.ComputeNode, crm pb_common.InternalComputeInfo) {
+
+// 	crm.Id = node.Id
+// 	crm.Name = node.Name
+// 	crm.DatapathIp = node.DatapathIp
+// 	crm.ContainerIp = node.ContainerIp
+// 	crm.Mac = node.Mac
+// 	crm.Veth = node.Veth
+// 	if node.Status == database.STATUS_READY {
+// 		crm.Status = pb_common.Status_READY
+// 	} else if node.Status == database.STATUS_DELETING {
+// 		crm.Status = pb_common.Status_DELETING
+// 	} else if node.Status == database.STATUS_DEPLOYING {
+// 		crm.Status = pb_common.Status_DEPLOYING
+// 	} else {
+// 		crm.Status = pb_common.Status_NONE
+// 	}
+// 	if node.OperationType == database.OPERATION_CREATE {
+// 		crm.OperationType = pb_common.OperationType_CREATE
+// 	} else if node.OperationType == database.OPERATION_INFO {
+// 		crm.OperationType = pb_common.OperationType_INFO
+// 	} else if node.OperationType == database.OPERATION_DELETE {
+// 		crm.OperationType = pb_common.OperationType_DELETE
+// 	} else if node.OperationType == database.OPERATION_UPDATE {
+// 		crm.OperationType = pb_common.OperationType_UPDATE
+// 	}
+
+// }
```

Comment:
@zzengcs please remove the commented-out code.

Reply:
resolved